src/test/java/com/users/service/UserServiceTest.java

```diff
@@ -0,0 +1,244 @@
+package com.users.service;
+
+import com.users.criteria.UserCriteriaEasy;
+import com.users.dto.UserDto;
+import com.users.entities.User;
+import com.users.exceptions.EntityAlreadyExistsException;
+import com.users.exceptions.EntityNotFoundException;
+import com.users.mappers.UserMapper;
+import com.users.repository.UserRepository;
+import com.users.utility.UserUtilityTest;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.junit.MockitoJUnitRunner;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.PageImpl;
+import org.springframework.data.domain.PageRequest;
+import org.springframework.data.domain.Pageable;
+import org.springframework.data.jpa.domain.Specification;
+
+import java.time.Instant;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Optional;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.anyLong;
+import static org.mockito.BDDMockito.given;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+@RunWith(MockitoJUnitRunner.class)
+public class UserServiceTest {
+
+    @Mock
+    private UserRepository userRepository;
+
+    @Mock
+    private UserMapper userMapper;
+
+    @InjectMocks
+    private UserService userService;
+
+    @Test
+    public void whenSaveUser_shouldReturnUser() {
+        List<UserDto> usersDto = UserUtilityTest.createUsersDto(1);
+        UserDto userDto = usersDto.get(0);
+
+        User user = new User();
+        user.setDateOfBirth(userDto.getDateOfBirth());
+        user.setPrimaryEmail(userDto.getPrimaryEmail());
+        user.setPhoneNumber(userDto.getPhoneNumber());
+
+        when(userMapper.toEntity(userDto)).thenReturn(user);
```

Comment:
aceste when uri ar trb sa reflecte ordinea din metoda, plus ca ai sarit 2 when uri, chiar daca tie nu ti dau eroare, dar mai apelezi de 2 ori repository in metoda.. deci when urile ar trb sa fie 
userRepository.existsByPrimaryEmail(user.getPrimaryEmail())
userRepository::existsByPrimaryEmail
userMapper.toEntity(user);
userRepository.save(userEntity);
userMapper.toDto(savedUser);